src/Configurator.php

```diff
@@ -169,6 +174,19 @@ public function withSetting($name, $value)
     }
 
     /**
+     * @param $extension string
+     * @param $className string
+     *
+     * @return Configurator
```

Comment:
Is `$this` a thing in doc comments? I was aware that `self` was preferable to the class name but I've not seen `$this` before. If it is a thing then I agree (and it should be applied to the other methods which return `$this` too) 😄